### DISCUSSION TITLE
Native support for GSM modules

### DISCUSSION THREAD

**Discussion by szerwi** [espressif/arduino-esp32#8922] [Q&A] [answered]
I would like to ask whether there is any support for GSM modules to access the Internet in `arduino-esp32`?
I found out that there is a network refactor ongoing: https://github.com/espressif/arduino-esp32/pull/8760
Does it mean, that after the refactor it will be possible to use GSM modules like SIM800/SIM7600 using common `WiFiClient` interface that is used for WiFi and Ethernet?

**Comment by me-no-dev** [comment, marked as answer]
After refactoring is done, support for PPPoS will be added. I can not estimate exact time that will be out, but should be in either 3.0.0 or 3.0.1
